cpp/ql/lib/semmle/code/cpp/rangeanalysis/new/internal/semantic/SemanticExprSpecific.qll

```diff
@@ -434,6 +434,50 @@ module SemanticExprConfig {
 
   /** Gets the expression associated with `instr`. */
   SemExpr getSemanticExpr(IR::Instruction instr) { result = Equiv::getEquivalenceClass(instr) }
+
+  private predicate typeBounds(SemType t, float lb, float ub) {
+    exists(SemIntegerType integralType, float limit |
+      integralType = t and limit = 2.pow(8 * integralType.getByteSize())
+    |
+      if integralType instanceof SemBooleanType
+      then lb = 0 and ub = 1
```

Comment:
I think, technically, this should be `0` and `-1`, or not? I also assume this was copied from the simple range analysis library?

Reply:
Indeed this was copied from simple range analysis. Why should we have `0` and `-1` here? AFAIK, the bounds should be inclusive.

Reply:
Ah, the standard says: "If the source type is bool, the value false is converted to zero and the value true is converted to one." So what you added seems correct.